pkg/datapath/linux/config/config.go

```diff
@@ -894,10 +894,6 @@ func (h *HeaderfileWriter) writeTemplateConfig(fw *bufio.Writer, e datapath.Endp
 		fmt.Fprintf(fw, "#define ENABLE_ROUTING 1\n")
 	}
 
-	if !e.DisableSIPVerification() {
-		fmt.Fprintf(fw, "#define ENABLE_SIP_VERIFICATION 1\n")
```

Comment:
I don't understand this change. Does that mean we won't define `ENABLE_SIP_VERIFICATION` by default anymore? What does the `node_config.h` file look like with this patch, if you leave everything to its default values.

Reply:
I'm not quite following. Could you give examples of what the generated config files look like before and after this patch with the problematic configuration?

Reply:
Without this patch, the first entry of "ENABLE_SIP_VERIFICATION" is written by following code. When a new pod/endpoint is created, if DatapathConfiguration.DisableSipVerification is true from the CNI, then "#define ENABLE_SIP_VERIFICATION 1" is not required in the ep_config.h, but if SourceIPVerification option is still true, then getFmtOpt() writes "#define ENABLE_SIP_VERIFICATION 1" to ep_config.h 

```
	if !e.DisableSIPVerification() {
		fmt.Fprintf(fw, "#define ENABLE_SIP_VERIFICATION 1\n")
```

Reply:
Sorry, this fell through the cracks of a busy week :disappointed: 

Doesn't the same issue affect every other runtime per-endpoint config? When you say "conflict", they are always defined with the exact same value so don't really conflict or is there a case where the two macro definitions could end up with different values?

Reply:
They could be different value as the case I described .
> When a new pod/endpoint is created, if DatapathConfiguration.DisableSipVerification is true from the CNI, then "#define ENABLE_SIP_VERIFICATION 1" is not required in the ep_config.h, but if SourceIPVerification option is still true, then getFmtOpt() writes "#define ENABLE_SIP_VERIFICATION 1" to ep_config.h